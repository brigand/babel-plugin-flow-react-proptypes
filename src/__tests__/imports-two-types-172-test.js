const babel = require('babel-core');
const content = `
import type { Foo, Bar } from './types';
var React = require('react');

type Props = {
  foo: Foo,
  bar: Bar,
}

const C = (props: Props) => <div />;
`;

it('imports-two-types-172', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: [['env', { modules: false }], 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatch(/\{\s*bpfrpt_proptype_Foo/);
  expect(res).toMatch(/\{\s*bpfrpt_proptype_Bar/);
  expect(res).toMatch(/typeof bpfrpt_proptype_Bar/);
  expect(res).toMatchSnapshot();
});
