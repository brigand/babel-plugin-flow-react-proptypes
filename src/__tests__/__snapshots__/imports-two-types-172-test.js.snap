// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imports-two-types-172 1`] = `
"var React = require('react');

var C = function C(props) {
  return /*#__PURE__*/React.createElement(\\"div\\", null);
};

C.propTypes = {
  \\"foo\\": function foo() {
    return (typeof bpfrpt_proptype_Foo === \\"function\\" ? bpfrpt_proptype_Foo.isRequired ? bpfrpt_proptype_Foo.isRequired : bpfrpt_proptype_Foo : PropTypes.shape(bpfrpt_proptype_Foo).isRequired).apply(this, arguments);
  },
  \\"bar\\": function bar() {
    return (typeof bpfrpt_proptype_Bar === \\"function\\" ? bpfrpt_proptype_Bar.isRequired ? bpfrpt_proptype_Bar.isRequired : bpfrpt_proptype_Bar : PropTypes.shape(bpfrpt_proptype_Bar).isRequired).apply(this, arguments);
  }
};
import { bpfrpt_proptype_Foo } from \\"./types\\";
import { bpfrpt_proptype_Bar } from \\"./types\\";
import PropTypes from \\"prop-types\\";"
`;
